test/ca/mcgill/cs/stg/jetuml/graph/TestUsageScenariosClassDiagram.java

```diff
@@ -0,0 +1,694 @@
+package ca.mcgill.cs.stg.jetuml.graph;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.awt.Graphics2D;
+import java.awt.geom.Point2D;
+import java.awt.geom.Rectangle2D;
+import java.awt.image.BufferedImage;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ca.mcgill.cs.stg.jetuml.diagrams.ClassDiagramGraph;
+import ca.mcgill.cs.stg.jetuml.framework.Clipboard;
+import ca.mcgill.cs.stg.jetuml.framework.GraphPanel;
+import ca.mcgill.cs.stg.jetuml.framework.Grid;
+import ca.mcgill.cs.stg.jetuml.framework.SelectionList;
+import ca.mcgill.cs.stg.jetuml.framework.ToolBar;
+
+/**
+ * Tests various interactions with Class Diagram normally triggered from the 
+ * GUI. Here we use the API to simulate GUI Operation for Class Diagram.
+ * 
+ * @author Jiajun Chen
+ *
+ */
+
+public class TestUsageScenariosClassDiagram 
+{
+	
+	private ClassDiagramGraph diagram;
+	private Graphics2D aGraphics;
+	private GraphPanel aPanel;
+	private Grid aGrid;
+	private Clipboard clipboard;
+	private SelectionList aList;
+	private ClassNode aClassNode = new ClassNode();
+	private InterfaceNode aInterfaceNode = new InterfaceNode();
+	private PackageNode aPackageNode = new PackageNode();
+	private NoteNode aNoteNode = new NoteNode();
+	private ClassRelationshipEdge aggrEdge = new AggregationEdge();
```

Comment:
please use complete nouns, e.g., `aggregationEdge`. Easy to do with rename refactoring.